File: src/main/java/icu/samnyan/aqua/sega/maimai2/handler/UserReqHandler.java
package icu.samnyan.aqua.sega.maimai2.handler;

import com.fasterxml.jackson.core.JsonProcessingException;
import icu.samnyan.aqua.sega.general.BaseHandler;

import java.util.Map;

public interface UserReqHandler extends BaseHandler {
    @Override
    default Object handle(Map<String, Object> request) throws JsonProcessingException {
        var userId = ((Number) request.get("userId")).longValue();
        return handleThis(request, userId);
    }

    Object handleThis(Map<String, Object> request, Long userId) throws JsonProcessingException;
}

File: src/main/java/icu/samnyan/aqua/sega/maimai2/handler/UserReqHandler.kt
package icu.samnyan.aqua.sega.maimai2.handler

import ext.parsing
import icu.samnyan.aqua.sega.general.BaseHandler

fun interface UserReqHandler : BaseHandler {
    override fun handle(request: Map<String, Any>): Any {
        val userId = parsing { (request["userId"] as Number).toLong() }
        return handleThis(request, userId)
    }

    fun handleThis(request: Map<String, Any>, userId: Long): Any
}
